### PR TITLE
Redirect user if branch is gone

### LINKED ISSUE #5432: Deleting PR branch after merging creates errors
# 🐛 bug report

## Preflight Checklist

- [ x] I have read the
      [Contributing Guidelines](https://github.com/codesandbox/codesandbox-client/blob/master/CONTRIBUTING.md)
      for this project.
- [x ] I agree to follow the
      [Code of Conduct](https://github.com/codesandbox/codesandbox-client/blob/master/CODE_OF_CONDUCT.md)
      that this project adheres to.
- [x ] I have searched the issue tracker for an issue that matches the one I want
      to file, without success.

## Description of the problem
Deleting the PR branch that the CodeSandbox project was referring to creates a weird state for the CodeSandbox project. 

## How has this issue affected you? What are you trying to accomplish?
Trying to work using PRs with main branch, instead of directly committing to main. I expect CodeSandbox to fall back to the main branch (or allow me to choose a branch), even unlinking with any github repo would work (as long as I can relink it again).

### To Reproduce
Link a CodeSandbox project to a GitHub repository.
Inside CodeSandbox, make changes.
Instead of committing directly to main, create a PR branch.
Merge PR branch using GitHub, delete branch afterwards.
CodeSandbox project cannot be linked anymore and is stuck :( 

Result: https://codesandbox.io/s/priceless-nightingale-y4m71

### Your Environment

| Software         | Name/Version |
| ---------------- | ------------ |
| Browser          | Chrome 88.0.4324.104 (Official Build) (64-bit)
| Operating System | Windows 10 Pro, OS Build 19041.746

### PR DESCRIPTION
**Pull request by SaraVieira**
CleanShot 2021-02-01 at 17 54 48

closes #5432

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/5433/commits/64ffb2a5be59a74a06f734ab15e794ee18a1176f is at https://pr5433.build.csb.dev/s/new.

File: packages/app/src/app/overmind/namespaces/git/actions.ts [hunk 1]
```diff
@@ -812,6 +812,11 @@ export const _compareWithSource: AsyncAction = async ({
     true
   );
 
+  if (!originalChanges) {
```
Comment (christianalfoni) [inline packages/app/src/app/overmind/namespaces/git/actions.ts:815]: I would rather put a `try/catch` around the call above to `compareGit`. That way the reason for it being there is closer to the modal logic and you do not have to deal with pesky "possibly null"

```ts
try {
const originalChanges = await effects.api.compareGit(
    sandbox.id,
    sandbox.originalGitCommitSha!,
    sandbox.originalGit!.branch,
     true
   );
   const updates = await actions.git._evaluateGitChanges(baseChanges.files);
   // and so on
} catch (error) {
  state.currentModal = 'notFoundBranchModal';
}
```

**Review by christianalfoni — CHANGES_REQUESTED** [review]
Unless I lack some context here I have a suggestion that would improve a little bit I think! But awesome work, this would definitely work 😄

File: packages/app/src/app/overmind/namespaces/git/actions.ts (hunk 1 above)
Comment (SaraVieira) [reply packages/app/src/app/overmind/namespaces/git/actions.ts:815]: That still showed the error that's why I did it this way

Comment (christianalfoni) [reply packages/app/src/app/overmind/namespaces/git/actions.ts:815]: Though did that error come from this? Or is there maybe some more logic running creating the error? What does the stack trace say? 😄

Comment (christianalfoni) [reply packages/app/src/app/overmind/namespaces/git/actions.ts:815]: Should return from the `catch` so no other logic keeps running. Gimme call tomorrow and we can take a look together 😄

Comment (SaraVieira) [reply packages/app/src/app/overmind/namespaces/git/actions.ts:815]: I'll call you arround eleven to explain better :)

**Review by christianalfoni — APPROVED** [review]
👍 Awesome 😄

**Comment by andrasbacsai** [comment]
@SaraVieira asked me to revert this one. She is not at the computer atm.

**Pull request merged**